packages/yarnpkg-fslib/sources/ZipFS.ts

```diff
@@ -714,6 +714,16 @@ export class ZipFS extends BasePortableFakeFS {
     if (resolvedP === `/`)
       return PortablePath.root;
 
+    const fileIndex = this.entries.get(resolvedP);
+    if (resolveLastComponent && fileIndex !== undefined) {
+      if (this.isSymbolicLink(fileIndex)) {
+        const target = this.getFileSource(fileIndex).toString() as PortablePath;
+        return ppath.resolve(ppath.dirname(resolvedP), target);
```

Comment:
Can be multiple levels deep (a symlink to a symlink):
```suggestion
        return this.resolveFilename(reason, ppath.resolve(ppath.dirname(resolvedP), target), resolveLastComponent);
```
(that's why the following part is a `while (true)`)

Reply:
Fixed and added a test